### PR TITLE
Iframes

### PR DESCRIPTION
**Pull request by JulianO911**

### CODE REVIEW

File: cypress/e2e/iframe.cy.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import {css} from "cypress/types/jquery";
```
Comment (holgiosalos) [inline cypress/e2e/iframe.cy.ts]: probably this is not required

File: cypress/e2e/iframe.cy.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    iframe = new Iframe();
+  });
+
+  it("testing the title after upload file", () => {
+    const htmlTitle = "HTML Tutorial";
+    const cssTitle = "CSS Tutorial";
+    iframe.visit();
+    iframe.getFrameTitle(htmlTitle);
+    iframe.goToCssPageInFrame(cssTitle);
+  });
```
Comment (holgiosalos) [inline cypress/e2e/iframe.cy.ts]: we should have two separated tests, you can put the visit call in the before hook. The tests are:

1. To verify the HTM Tutorial title
2. To verify the CSS Tutorial title

File: cypress/page/iframe.page.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        this.iFrameCssButtonSelector = "a[title='CSS Tutorial']"
+    }
+
+    public visit(): void{
+        cy.visit(this.URL)
+      }
+      
+    public getFrameTitle(titleElements: string): void{
+        cy.iframe().find(this.iFrameSelector).should("contains.text",titleElements)
+      }
```
Comment (holgiosalos) [inline cypress/page/iframe.page.ts]: in this case, as we are asking to call the method as get, it should not perform any validation, you just should return the title of the iframe

File: cypress/page/iframe.page.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      
+    public  goToCssPageInFrame(titleElements: string): void{
+        cy.intercept(
+          this.URL,
+          {
+            delay: 1000,
+          }
+        ).as('getDelayedStories')
```
Comment (holgiosalos) [inline cypress/page/iframe.page.ts]: this intercept is not required

File: cypress/page/iframe.page.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    public visit(): void{
+        cy.visit(this.URL)
+      }
+      
+    public getFrameTitle(titleElements: string): void{
+        cy.iframe().find(this.iFrameSelector).should("contains.text",titleElements)
+      }
+      
+    public  goToCssPageInFrame(titleElements: string): void{
```
Comment (holgiosalos) [inline cypress/page/iframe.page.ts]: we don't expect to have an input parameter in this method

File: cypress/page/iframe.page.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+class Iframe{
+    private URL: string;
+    private iFrameSelector: string;
+    private iFrameCssButtonSelector: string;
+
+    constructor(){
+        this.URL = "https://www.w3schools.com/html/html_iframe.asp"
+        this.iFrameSelector = "#main h1"
```
Comment (holgiosalos) [inline cypress/page/iframe.page.ts]: I think that this selector is not for the iframe but for the title, so, it should be named as `frameTitle` or something similar

File: cypress/page/iframe.page.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        this.iFrameCssButtonSelector = "a[title='CSS Tutorial']"
+    }
+
+    public visit(): void{
+        cy.visit(this.URL)
+      }
+      
+    public getFrameTitle(titleElements: string): void{
+        cy.iframe().find(this.iFrameSelector).should("contains.text",titleElements)
```
Comment (holgiosalos) [inline cypress/page/iframe.page.ts]: to call the iframe command you need to specify a selector inside.

File: cypress/page/iframe.page.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
+class Iframe{
+    private URL: string;
+    private iFrameSelector: string;
+    private iFrameCssButtonSelector: string;
+
+    constructor(){
+        this.URL = "https://www.w3schools.com/html/html_iframe.asp"
+        this.iFrameSelector = "#main h1"
+        this.iFrameCssButtonSelector = "a[title='CSS Tutorial']"
```
Comment (holgiosalos) [inline cypress/page/iframe.page.ts:10]: a better name for this would be cssTutorialLink or similar

File: cypress/page/iframe.page.ts (hunk 3 above)
Comment (JulianO911) [reply cypress/page/iframe.page.ts]: should i print it in cypress interface? Or just return it and store it in a variable?

**Review by holgiosalos — APPROVED** [review]
Congrats @JulianO911! you are the third or fourth in finishing the workshop!

**Pull request merged**